config/default/jenkins-infra.yaml

```diff
@@ -149,104 +150,48 @@ jenkins:
-                      scanCredentialsId('github-access-token')
-                      repoOwner('jenkins-infra')
-                      repository('docker-builder')
-                    }
-                  }
-                  factory {
-                    workflowBranchProjectFactory {
-                      scriptPath('Jenkinsfile_k8s')
+                [
+                  [repo: 'docker-packer', owner: 'jenkins-infra', name: 'Packer Docker Build'],
+                  [repo: 'docker-mirrorbits', owner: 'jenkins-infra', name: 'Mirrorbits Docker Build'],
+                  [repo: 'docker-datadog', owner: 'jenkins-infra', name: 'Datadog Docker Build'],
+                  [repo: 'docker-terraform', owner: 'jenkins-infra', name: 'Terraform Docker Build'],
+                  [repo: 'docker-builder', owner: 'jenkins-infra', name: "Docker Builder's Docker Build"],
+                  [repo: 'aws', owner: 'jenkins-infra', name: 'AWS Infra'],
+                ].each{ config ->
+                  multibranchPipelineJob(config.repo) {
+                    displayName config.name
+                    branchSources {
+                      github {
+                        id('2020122201') // not sure if the id is needed
+                        scanCredentialsId('github-access-token')
+                        repoOwner(config.owner)
+                        repository(config.repo)
+                      }
                     }
-                  }
-                }
-            - script: >
-                multibranchPipelineJob('aws') {
-                  displayName "AWS Infra"
-                  branchSources {
-                    github {
-                      id('2020120401')
-                      scanCredentialsId('github-access-token')
-                      repoOwner('jenkins-infra')
-                      repository('aws')
+                    factory {
+                      workflowBranchProjectFactory {
+                        scriptPath('Jenkinsfile_k8s')
+                      }
                     }
-                  }
-                  factory {
-                    workflowBranchProjectFactory {
-                      scriptPath('Jenkinsfile_k8s')
+                    configure { node ->
+                      def traits = node / 'sources' / 'data' / 'jenkins.branch.BranchSource' / 'source' / 'traits'
+                      traits << 'org.jenkinsci.plugins.github__branch__source.BranchDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.OriginPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                      }
+                      traits << 'org.jenkinsci.plugins.github__branch__source.ForkPullRequestDiscoveryTrait' {
+                        strategyId(1)
+                        trust(class: 'org.jenkinsci.plugins.github_branch_source.ForkPullRequestDiscoveryTrait$TrustNobody')
```

Comment:
why? this makes it so that Jenkinsfile changes can't be tested by anyone from fork PRs afaik

Reply:
changed to TrustPermission

Reply:
and made it configurable per repo